### PR TITLE
docs: add all missing docs

### PR DESCRIPTION
**Pull request by hamada147**
# Overview

Fixes #<issue number>

### My PR contains...
* [x] No code changes (changes to documentation, CI, metadata, etc.)
* [ ] Bug fixes (non-breaking change which fixes an issue)
* [ ] Improvements (misc. changes to existing features)
* [ ] Features (non-breaking change which adds functionality)

### My changes...
* [ ] are breaking changes
* [x] are not breaking changes
* [ ] If yes to above: I have updated the documentation accordingly

### Documentation
* [ ] My changes do not require a change to the project documentation
* [x] My changes require a change to the project documentation
* [ ] If yes to above: I have updated the documentation accordingly

### Tests
* [x] My changes can not or do not need to be tested
* [ ] My changes can and should be tested by unit and/or integration tests
* [ ] If yes to above: I have added tests to cover my changes
* [ ] If yes to above: I have taken care to cover edge cases in my tests

### CODE REVIEW

**Comment by atala-dev** [comment]
### Code Coverage
|File|Coverage [78.39%]|:x:|
|:-|:-:|:-:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/Platform.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2FPlatform.kt)|0.00%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/hashing/PBKDF2SHA512.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fhashing%2FPBKDF2SHA512.kt)|50.00%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/secp256k1/Secp256k1Lib.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecp256k1%2FSecp256k1Lib.kt)|86.67%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/securerandom/SecureRandom.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecurerandom%2FSecureRandom.kt)|100.00%|:white_check_mark:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/JVMBigIntegerExt.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FJVMBigIntegerExt.kt)|0.00%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdKeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdKeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPrivateKey.kt)|66.67%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPublicKey.kt)|77.78%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519KeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519KeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PublicKey.kt)|100.00%|:white_check_mark:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/secp256k1/Secp256k1Lib.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecp256k1%2FSecp256k1Lib.kt)|86.67%|:x:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/securerandom/SecureRandom.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecurerandom%2FSecureRandom.kt)|100.00%|:white_check_mark:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdKeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdKeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPrivateKey.kt)|66.67%|:x:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPublicKey.kt)|77.78%|:x:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519KeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519KeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PublicKey.kt)|100.00%|:white_check_mark:|
|[apollo/src/iosMain/kotlin/io/iohk/atala/prism/apollo/Platform.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FiosMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2FPlatform.kt)|0.00%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/Platform.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2FPlatform.kt)|0.00%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/secp256k1/Secp256k1Lib.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecp256k1%2FSecp256k1Lib.kt)|86.67%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/securerandom/SecureRandom.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecurerandom%2FSecureRandom.kt)|100.00%|:white_check_mark:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/ByteArrayExt.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FByteArrayExt.kt)|28.57%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdKeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdKeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPrivateKey.kt)|66.67%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPublicKey.kt)|77.78%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519KeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519KeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PublicKey.kt)|100.00%|:white_check_mark:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/Platform.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2FPlatform.kt)|0.00%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/secp256k1/Secp256k1Lib.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecp256k1%2FSecp256k1Lib.kt)|86.67%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/securerandom/SecureRandom.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Fsecurerandom%2FSecureRandom.kt)|100.00%|:white_check_mark:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/JVMBigIntegerExt.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FJVMBigIntegerExt.kt)|0.00%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdKeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdKeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPrivateKey.kt)|66.67%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMEdPublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMEdPublicKey.kt)|77.78%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519KeyPair.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519KeyPair.kt)|100.00%|:white_check_mark:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PublicKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PublicKey.kt)|100.00%|:white_check_mark:|
|[apollo/src/macosMain/kotlin/io/iohk/atala/prism/apollo/Platform.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/b8d0ffd2c16c0661e78acf7cf09a3d1f518d6fe4/apollo%2Fsrc%2FmacosMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2FPlatform.kt)|0.00%|:x:|

|Total Project Coverage|91.66%|:white_check_mark:|
|:-|:-:|:-:|

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt [hunk 1]
```diff
@@ -67,6 +90,9 @@ open external class SHA512 : SHA2<SHA512> {
     override fun destroy()
 }
 
+/**
+ * The sha512 class provides a platform-specific implementation for generating SHA-512 hash.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt:94]:
```suggestion
 * The SHA512 class provides a platform-specific implementation for generating SHA-512 hash.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt [hunk 2]
```diff
@@ -127,6 +226,15 @@ internal external interface Sha384 : BlockHash<Sha384>, MessageDigest<Sha384> {
     override var endian: String /* "big" */
 }
 
+/**
+ * Sha512 is an internal external interface that extends both BlockHash and MessageDigest interfaces.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:230]:
```suggestion
 * SHA512 is an internal external interface that extends both BlockHash and MessageDigest interfaces.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt [hunk 3]
```diff
@@ -127,6 +226,15 @@ internal external interface Sha384 : BlockHash<Sha384>, MessageDigest<Sha384> {
...
+ * It represents the SHA-512 hash algorithm.
+ *
+ * Please note that this documentation only focuses on the specific properties and methods provided by the Sha512 interface.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:233]:
```suggestion
 * Please note that this documentation only focuses on the specific properties and methods provided by the SHA512 interface.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt [hunk 4]
```diff
@@ -103,6 +190,12 @@ internal external interface Sha1 : BlockHash<Sha1>, MessageDigest<Sha1> {
     override var endian: String /* "big" */
 }
 
+/**
+ * The `Sha224` class represents the SHA-224 hash algorithm. It implements the `BlockHash` and `MessageDigest` interfaces.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:194]:
```suggestion
 * The `SHA224` class represents the SHA-224 hash algorithm. It implements the `BlockHash` and `MessageDigest` interfaces.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt (hunk 2 above)
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:230]:
```suggestion
 * SHA512 is an internal external interface that extends both BlockHash and MessageDigest interfaces.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt [hunk 5]
```diff
@@ -47,46 +86,86 @@ internal external interface ShaSet {
...
+/**
+ * An external interface representing a constructor for the SHA256 algorithm.
+ */
 internal external interface Sha256Constructor {
     @nativeInvoke
     operator fun invoke(): Sha256
 }
 
+/**
+ * The Sha384Constructor interface represents a constructor function for the Sha384 interface,
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:140]:
```suggestion
 * The Sha384Constructor interface represents a constructor function for the SHA384 interface,
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt [hunk 6]
```diff
@@ -47,46 +86,86 @@ internal external interface ShaSet {
...
+ * which is an implementation of the BlockHash and MessageDigest interfaces.
+ */
 internal external interface Sha384Constructor {
     @nativeInvoke
     operator fun invoke(): Sha384
 }
 
+/**
+ * The `Sha512Constructor` interface represents a constructor function for creating instances of the `Sha512` interface.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:149]:
```suggestion
 * The `Sha512Constructor` interface represents a constructor function for creating instances of the `SHA512` interface.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt (hunk 4 above)
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/HashJS.kt:194]:
```suggestion
 * The `SHA224` class represents the SHA-224 hash algorithm. It implements the `BlockHash` and `MessageDigest` interfaces.
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt [hunk 7]
```diff
@@ -83,6 +111,8 @@ external object sha512_224 {
     fun create(): Any
 }
 
+/**
+ * The `sha512_256` class represents an object that can compute SHA-512*/
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt:115]:
```suggestion
 * The `SHA512_256` class represents an object that can compute SHA-512*/
```

File: apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt [hunk 8]
```diff
@@ -91,6 +121,9 @@ external object sha512_256 {
     fun create(): Any
 }
 
+/**
+ * The `sha384` class provides functions for computing the SHA-384 hash.
```
Comment (petevielhaber) [inline apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/hashing/external/SHA512.kt:125]:
```suggestion
 * The `SHA384` class provides functions for computing the SHA-384 hash.
```

**Review by petevielhaber — APPROVED** [review]
